langs/python.go

```diff
@@ -71,17 +71,17 @@ func (h *PythonLangHelper) Entrypoint() (string, error) {
 }
 
 func (h *PythonLangHelper) DockerfileBuildCmds() []string {
-	pip := "pip3"
 	r := []string{}
-	r = append(r, "RUN apt-get update && apt-get install --no-install-recommends -qy build-essential gcc")
+	r = append(r, "ADD . /function/")
 	if exists("requirements.txt") {
-		r = append(r,
-			"ADD requirements.txt /function/",
-			fmt.Sprintf("RUN %v install --no-cache --no-cache-dir -r requirements.txt", pip),
-		)
+		r = append(r, `
+RUN apt-get update && apt-get install --no-install-recommends -qy build-essential gcc &&\
```

Comment:
I think this change is fine now but I think the python FDK build itself is wrong to be running  apt commands by default during fn builds - I've raised this to track: https://github.com/fnproject/fdk-python/issues/44

Reply:
The way this change was introduced makes the whole build process slower than it was before. With this change package install process will be triggered with any change to *.py files, but it’s wrong. The only reason to trigger package install process is change to requirements.txt

So, this change has to be rewritten to reduce build time as it was before, but with new purge thing.

Reply:
I believe proper way will be just to create apt repo and have pre-built package like python3-fdk for stretch. In that case no compile or other installed packages will be required during build. Or in ideal case base image shall have FDK installed so build process will just put func.py to image and install custom requirements from requirements.txt if any. And even in that case you need to have build essentials only for native pip requirements, pure python packages does not need it.

Reply:
I tend to disagree with you because build packages can be required by other dependencies that require compiling (tensorflow, numpy, etc.)

Reply:
They also can require devel packages too (like openssl-devel), so keeping just compiler will not help anyway

Reply:
I think best solution from size point of view - multistage images, so release image will not have unnecessary things. But image creation time will be much longer